File: MealMaster/MealMaster.BLL/DTOs/Response/Recipe/RecipeResponseDto.cs
using MealMaster.BLL.DTOs.Response.CuisineType;
using MealMaster.BLL.DTOs.Response.DietaryRestriction;
using MealMaster.BLL.DTOs.Response.Product;

namespace MealMaster.BLL.DTOs.Response.Recipe;

public class RecipeResponseDto
{
    public Guid UserId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int Calories { get; set; }
    public int PreparationTime { get; set; }
    public CuisineTypeResponseDto CuisineType { get; set; }
    public DateTime CreatedDate { get; set; }
    public DietaryRestrictionResponseDto Restriction { get; set; }
    public List<ProductResponseDto> Products { get; set; }
}

File: Client/meal-master/src/app/models/recipe-response.dto.ts
// src/app/models/recipe-response.dto.ts
import { ProductResponseDto } from './product-response.dto';
import { CuisineTypeResponseDto } from './cuisine-type-response.dto';
import { DietaryRestrictionResponseDto } from './dietary-restriction-response.dto';

export interface RecipeResponseDto {
  id: string;
  userId: string;
  name: string;
  description: string;
  calories: number;
  preparationTime: number;
  cuisineType: CuisineTypeResponseDto;
  createdDate: string; // ISO Date string
  restriction: DietaryRestrictionResponseDto;
  products: ProductResponseDto[];
}
